acceptanceTests/cypress/integration/tests/UiTest.js

```diff
@@ -0,0 +1,213 @@
+    cy.get(`entry-${entryNum}`).should('not.exist');
+}
+
+function checkIllegalFilter(illegalFilterName) {
+    it(`should show red search bar with the input: ${illegalFilterName}`, function () {
+        cy.get('#total-entries').then(number => {
+            const totalEntries = number.text();
+
+            cy.get('.w-tc-editor-text').type(illegalFilterName);
+            cy.get('.w-tc-editor').should('have.attr', 'style').and('include', redFilterColor);
+            cy.get('[type="submit"]').click();
+
+            cy.get('[role="alert"]').should('be.visible');
+            cy.get('.w-tc-editor-text').clear();
+
+            cy.reload();
+            cy.get('#total-entries', {timeout: 10000}).should('have.text', totalEntries);
+        });
+    });
+}
+function checkFilter(filterDetails){
+    const {name, leftSidePath, rightSidePath, rightSideExpectedText, leftSideExpectedText, applyByEnter} = filterDetails;
+
+    it(`checking the filter: ${name}`, function () {
+        cy.get('#total-entries').then(number => {
+            const totalEntries = number.text();
+
+            // checks the hover on the last entry (the only one in DOM at the beginning)
+            leftOnHoverCheck(totalEntries - 1, leftSidePath, name);
+
+            // applying the filter with alt+enter or with the button
+            cy.get('.w-tc-editor-text').type(`${name}${applyByEnter ? '{alt+enter}' : ''}`);
+            cy.get('.w-tc-editor').should('have.attr', 'style').and('include', greenFilterColor);
+            if (!applyByEnter)
+                cy.get('[type="submit"]').click();
+
+            // still one entry only, so checking all four checks on it
+            leftTextCheck(totalEntries - 1, leftSidePath, leftSideExpectedText);
+            leftOnHoverCheck(totalEntries - 1, leftSidePath, name);
+            rightTextCheck(rightSidePath, rightSideExpectedText);
+            rightOnHoverCheck(rightSidePath, name);
+
+            cy.get('[title="Fetch old records"]').click();
+            cy.viewport(1920, 3500); // resizing the window so all entries will be in DOM
+
+            cy.get('#entries-length', {timeout: 10000}).should('have.text', totalEntries);
+
+            // checking only 'leftTextCheck' on all entries because the rest of the checks require more time
+            [...Array(parseInt(totalEntries)).keys()].forEach(entryNum => {
+                leftTextCheck(entryNum, leftSidePath, leftSideExpectedText);
+            });
+
+            // making the other 3 checks on the first five entries (longer time for each check)
+            firstFiveDeeperChcek(leftSidePath, rightSidePath, name, leftSideExpectedText, rightSideExpectedText);
+
+            cy.viewport(1920, 1080);
+            cy.reload();
+            cy.get('#total-entries', {timeout: 10000}).should('have.text', totalEntries);
+
+        });
+    });
+}
+
+function firstFiveDeeperChcek(leftSidePath, rightSidePath, filterName, leftSideExpectedText, rightSideExpectedText) {
```

Comment:
pass the number of entries to check as a param, so it will be easier to change in the future